tasks/utils.py

```diff
@@ -147,9 +147,16 @@ def get_version_ldflags(ctx, prefix=None):
     payload_v = get_payload_version()
     commit = get_git_commit()
 
+    agent_version = get_version(ctx, include_git=True, prefix=prefix)
+
     ldflags = "-X {}/pkg/version.Commit={} ".format(REPO_PATH, commit)
-    ldflags += "-X {}/pkg/version.AgentVersion={} ".format(REPO_PATH, get_version(ctx, include_git=True, prefix=prefix))
+    ldflags += "-X {}/pkg/version.AgentVersion={} ".format(REPO_PATH, agent_version)
     ldflags += "-X {}/pkg/serializer.AgentPayloadVersion={} ".format(REPO_PATH, payload_v)
+
+    # If we're building Agent 7, we want to force the use of Python 3
+    if agent_version.startswith("7"):
```

Comment:
I'm not sure this should be the test. Instead of doing this only for Agent 7, I think this ldflag should be set whenever we build the Agent with only one (or no) version of python.

Maybe this variable passed to the main invoke task should be the one we use? https://github.com/DataDog/datadog-agent/blob/f491bfb4e140eba8a7b5b30c5da66e8246ce511b/tasks/agent.py#L170

(also, IMO this ldflag shouldn't be set in `get_version_ldflags` if you agree with me that the logic shouldn't rely on the actual agent version)

Reply:
I preferred to test `7` as we don't know the strategy for the next major and relying on `with_both_python` is slightly different. But it's not a strong opinion so I'll change that.

Good one for not being in `get_version_ldflags`.

Reply:
That should be good, lmk.